processor/spanmetricsprocessor/processor.go

```diff
@@ -332,19 +332,15 @@ func (p *processorImp) updateLatencyMetrics(key metricKey, latency float64, inde
 	p.latencyBucketCounts[key][index]++
 }
 
-func buildDimensionKVs(serviceName string, span pdata.Span, optionalDims []Dimension) dimKV {
+func buildDimensionKVs(serviceName string, span pdata.Span, optionalDims []Dimension, resourceAttrs pdata.AttributeMap) dimKV {
 	dims := make(dimKV)
 	dims[serviceNameKey] = serviceName
 	dims[operationKey] = span.Name()
 	dims[spanKindKey] = span.Kind().String()
 	dims[statusCodeKey] = span.Status().Code().String()
-	spanAttr := span.Attributes()
 	for _, d := range optionalDims {
-		if attr, ok := spanAttr.Get(d.Name); ok {
-			dims[d.Name] = tracetranslator.AttributeValueToString(attr)
-		} else if d.Default != nil {
-			// Set the default if configured, otherwise this metric should have no value set for the dimension.
-			dims[d.Name] = *d.Default
+		if v, ok := getDimensionValue(d, span, resourceAttrs); ok {
+			dims[d.Name] = v
```

Comment:
Not sure if it would be a good idea to print a `DEBUG` log in case of missing attribute and no `default` is set.

Reply:
Thanks @hossain-rayhan, I've added a debug log which looks like:
```
 "foo" metric dimension omitted; not found and no default configured     {"kind": "processor", "name": "spanmetrics", "service.name": "currencyservice", "operation": "/GetConversion", "span.kind": "SPAN_KIND_SERVER", "status.code": "STATUS_CODE_UNSET"}
```

I added the extra structured fields to the log message so it's easier to identify the spans that are missing these attributes, traded off for more verbose logging.

Reply:
reason: I figured there's not much benefit to adding debug logging unless if there's enough information to address the problem (if it is a problem).